src/main/java/org/opencraft/server/task/impl/CreeperTask.java

```diff
@@ -0,0 +1,94 @@
+ *       this list of conditions and the following disclaimer.
+ *
+ *     * Distributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *
+ *     * Neither the name of the OpenCraft nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.opencraft.server.task.impl;
+
+import org.opencraft.server.game.impl.CTFGameMode;
+import org.opencraft.server.game.impl.GameSettings;
+import org.opencraft.server.model.Player;
+import org.opencraft.server.model.Position;
+import org.opencraft.server.model.Level;
+import org.opencraft.server.model.World;
+import org.opencraft.server.task.ScheduledTask;
+
+/* Does the delayed explosion for creeper item. */
+public class CreeperTask extends ScheduledTask {
+  private static final boolean LETHAL = true;
+  private static final boolean TEAMKILL = true;
+  private static final String NAME = "Creeper";
+
+  private CTFGameMode ctf;
+  private Player invoker;
+  private int pastDeaths;
+  private Level level;
+
+  public CreeperTask(Player invoker, Level level) {
+    super(0);
+
+    this.ctf = (CTFGameMode)World.getWorld().getGameMode();
+    this.invoker = invoker;
+    this.pastDeaths = invoker.deaths;
+    this.level = level;
+
+    float creeperTime = GameSettings.getFloat("CreeperTime");
+    if (creeperTime > 0.0f) {
+      this.setDelay((long)(1000 * creeperTime));
+    }
+  }
+
+  public void execute() {
+    // Do not explode if the player died.
+    if (pastDeaths == invoker.deaths) {
+      doCreeper();
+    }
+  }
+
+  private void doCreeper() {
+    int radius = GameSettings.getInt("CreeperRadius");
+    if (radius < 0) {
+      radius = 0;
+    }
+
+    Position pos = this.invoker.getPosition().toBlockPos();
+
+    ctf.explodeTNT(
+      this.invoker, this.level, pos.getX(), pos.getY(), pos.getZ(),
+      radius, LETHAL, TEAMKILL, false, NAME
+    );
+
+    this.stop();
```

Comment:
Becuse ScheduledTask is for tasks that are meant to be repeated throughout the lifetime of the server, and this only calls stop on doCreeper, If you kill somebody who has run /cr before it goes off, the creeper task never exits and executes infinitely until the server restarts, causing resource leaks.

Reply:
My bad, the task should always be stopped regardless of whether the explosion happens or not.